MatrixSDKTests/MXBackgroundSyncServiceTests.swift

```diff
@@ -584,6 +584,163 @@ class MXBackgroundSyncServiceTests: XCTestCase {
+    // - Bob uses the MXBackgroundSyncService to fetch it
+    // - Alice sends a message. This make bob MXSession update its sync token
+    // - Bob uses the MXBackgroundSyncService again
+    // -> MXBackgroundSyncService should have detected that the MXSession ran in parallel.
+    //    It must have reset its cache. syncResponseStore.prevBatch must not be the same
+    // -> MXSyncResponseFileStore must be in sync with MXSession store
+    func testWithMXSessionRunningInParallel() {
+        
+        // - Alice and Bob are in an encrypted room
+        let aliceStore = MXMemoryStore()
+        let bobStore = MXFileStore()
+        e2eTestData.doE2ETestWithAliceAndBob(inARoom: self, cryptedBob: true, warnOnUnknowDevices: false, aliceStore: aliceStore, bobStore: bobStore) { (aliceSession, bobSession, roomId, expectation) in
+            
+            guard let roomId = roomId, let room = aliceSession?.room(withRoomId: roomId) else {
+                XCTFail("Cannot set up initial test conditions - error: room cannot be retrieved")
+                expectation?.fulfill()
+                return
+            }
+            
+            guard let bobCredentials = bobSession?.credentials else {
+                XCTFail("Cannot set up initial test conditions - error: Bob's credentials cannot be retrieved")
+                expectation?.fulfill()
+                return
+            }
+            
+            let bobSessionStartEventStreamToken1 = bobSession!.store.eventStreamToken
+                
+            // - Alice sends a message
+            var localEcho: MXEvent?
+            room.sendTextMessage(Constants.messageText, localEcho: &localEcho) { (response) in
+                switch response {
+                    case .success(let eventId):
+                        
+                        guard let eventId = eventId else {
+                            XCTFail("Cannot set up initial test conditions - error: room cannot be retrieved")
+                            expectation?.fulfill()
+                            return
+                        }
+                        
+                        // - Bob uses the MXBackgroundSyncService to fetch it
+                        self.bgSyncService = MXBackgroundSyncService(withCredentials: bobCredentials)
+                        self.bgSyncService?.event(withEventId: eventId, inRoom: roomId) { _ in
+                            
+                            let syncResponseStore = MXSyncResponseFileStore()
+                            syncResponseStore.open(withCredentials: bobCredentials)
+                            let syncResponseStorePrevBatch = syncResponseStore.prevBatch
+                            
+                            // Test check
+                            // MXSyncResponseFileStore should point to the same sync token as MXSession store
+                            XCTAssertNotEqual(bobSessionStartEventStreamToken1, syncResponseStorePrevBatch)
```

Comment:
This should be `XCTAssertEqual` in theory, but this is worthless. Because Bob's session is not stopped in this test, we cannot guarantee that his store's eventStreamToken will be same as sync response store's prevBatch.